libethcore/SealEngine.cpp

```diff
@@ -55,6 +56,18 @@ void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, Transactio
 
 	if (_header.number() >= chainParams().homesteadForkBlock && (_ir & ImportRequirements::TransactionSignatures) && _t.hasSignature())
 		_t.checkLowS();
+
+    eth::EVMSchedule const& schedule = evmSchedule(_header.number());
+
+    // Pre calculate the gas needed for execution
+    if (_t.baseGasRequired(schedule) > _t.gas())
+        BOOST_THROW_EXCEPTION(OutOfGasIntrinsic() << RequirementError(
+                                  (bigint)(_t.baseGasRequired(schedule)), (bigint)_t.gas()));
+
+    // Avoid transactions that would take us beyond the block gas limit.
+    if (_gasUsed + (bigint)_t.gas() > _header.gasLimit())
+        BOOST_THROW_EXCEPTION(BlockGasLimitReached() << RequirementError(
+                                  (bigint)(_header.gasLimit() - _gasUsed), (bigint)_t.gas()));
```

Comment:
The error should contain the compared two values above
`_gasUsed + (bigint)_t.gas()` and `_header.gasLimit()`.

Reply:
requirenment error says that transaction gas should have been not more then 
`gasLimit - gasUsed`
but it got `t.gas` instead

the if could be 
```
if (_t.gas() > _header.gasLimit() - _gasUsed)
```
maybe it was written like this. because  of the bigint ?

Reply:
I'd keep the `if` condition as it is, and change the exception.

Reply:
to what?

Reply:
```
required:  block gasLimit 
but got:    blockGasUsed + t.gas() 
```
?